uaa/src/test/java/org/cloudfoundry/identity/uaa/orchestrator/OrchestratorClaimAT.java

```diff
@@ -1,124 +1,343 @@
+            return true;
+        } catch (ServiceInstanceProviderException | ApiException e) {
+            LOGGER.error("UAA Claim create failed for claimName: {}, uaaRequest:{}, errorMessage: {}",
+                    claimName, uaaRequest, e.getMessage(), e);
             throw e;
         }
     }
 
-    private Claim getZone(final String claimName) throws ServiceInstanceProviderException {
+    private Claim getZone(final String claimName, final ServiceName serviceName) throws ServiceInstanceProviderException {
         try {
-            return orchestratorUtil.readClaim(claimName, ServiceName.UAA);
+            return orchestratorUtil.readClaim(claimName, ServiceName.UAA, ApiEnvironment.valueOf(environment));
         } catch (ServiceInstanceProviderException e) {
             LOGGER.error("UAA Claim read failed for claimName: {}, errorMessage: {}", claimName,
-                         e.getMessage(), e);
+                    e.getMessage(), e);
             throw e;
         }
     }
 
-    private boolean deleteZone(final String claimName) throws ServiceInstanceProviderException {
+
+    private boolean deleteZone(final String claimName, final ServiceName serviceName) throws ServiceInstanceProviderException {
         try {
-            orchestratorUtil.deleteClaim(claimName, ServiceName.UAA);
+            orchestratorUtil.deleteClaim(claimName, ServiceName.UAA, ApiEnvironment.valueOf(environment));
             return true;
         } catch (ServiceInstanceProviderException e) {
             LOGGER.error("UAA Claim delete failed for claimName: {}, errorMessage: {}", claimName,
-                         e.getMessage(), e);
+                    e.getMessage(), e);
             throw e;
         }
     }
+
+    private String getZoneByName(final String zoneName, HttpStatus status, String expectedErrorMessage) throws Exception {
+        if (StringUtils.isEmpty(zoneName)) {
+            throw new Exception("zoneName can not be null");
+        }
+        String url = uaaUrl + ZONES_URI + "?name=" + zoneName;
+
+        return executeRequest(getTrustedIssuerValidScopesRestTemplate(), url, HttpMethod.GET, headers, status, "Zone[iamqa-svc-"+claimName+"] not found.",
+                null);
+    }
+
+    public static String executeRequest(OAuth2RestOperations tenantRestTemplate, String url, HttpMethod method,
+                                        HttpHeaders headers, HttpStatus expectedStatusCode,
+                                        String expectedMessage, String requestBody) {
+        String body = null;
+        if (!HttpMethod.GET.equals(method)) {
+            body = requestBody;
+        }
+        HttpEntity<String> request = new HttpEntity<>(body, headers);
+        log.trace("Executing HTTP {} - url: {}, request:\n {}", method, url, request);
+        if (!expectedStatusCode.is2xxSuccessful()) {
+            ResponseEntity<RestErrorResponse> responseEntity = tenantRestTemplate.exchange(url, method, request,
```

Comment:
Use [OrchestratorZoneResponse](https://github.com/GESoftware-CF/uaa/blob/predix_extensions_75.18.1/zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/model/OrchestratorZoneResponse.java) instead of RestErrorResponse

Reply:
Done